miqi/runtime/turn_runner.py

```
@@ -429,9 +551,51 @@ async def _drain_steer_messages() -> list[dict[str, Any]]:
+                import json as _json
+
+                from miqi.agent.tools.todo_write import TodoWriteTool
+
+                run_ctx = getattr(turn, "_run_ctx", None)
+                for tc in todo_calls:
+                    tool = TodoWriteTool(run_ctx.todo_state if run_ctx is not None else None)
+                    try:
+                        raw_args = getattr(tc, "arguments", None)
+                        args = _json.loads(raw_args) if isinstance(raw_args, str) and raw_args else (raw_args or {})
+                        result_text = await tool.execute(**args)
+                    except Exception as exc:  # pragma: no cover - defensive
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)